library/sha256.c

```diff
@@ -167,20 +167,23 @@ void mbedtls_sha256_clone( mbedtls_sha256_context *dst,
  */
 int mbedtls_sha256_starts( mbedtls_sha256_context *ctx, int is224 )
 {
-#if defined(MBEDTLS_SHA224_C)
```

Comment:
Out of scope for this PR, but I think that at this point having separate `sha224` function for at least `starts`, would help a lot. Majority of logic for this function is just handling the `is224`-related variables.

Reply:
I agree with you, this is a good idea. Besides I think it also emphasizes the independence of sha224/sha384 from sha256/sha512. No need for an extra PR to implement this: I'll do it in this one ;)

Reply:
I have one question though: apart from the `starts` functions, wouldn't it be better to rename also other functions (at least the ones used in the public interface) from `mbedtls_sha512_*` to `mbedtls_sha512_sha384_*`?
Otherwise it might happen that you call the `mbedtls_sha384_starts()`, but then use `mbedtls_sha512_*` functions...

Reply:
We can't just rename things: functions that exist now must continue to exist unchanged until the next major version (so, Mbed TLS 4.0) at least. So, we'd need to have both the old and the new functions for the transition period.

I see your point, but I think this is not something we want to be doing at this point: those modules are going to become purely internal (no longer part of the public API) in the next major version, and in the long run only the PSA Crypto API will remain for computing hashes. So, we don't want to spend time improving the APIs for these low-level modules.

Even if we did, I'd say it should be a separate PR, to avoid scope creep: you start wanting to add just a new `starts()` function, you end up wanting to rename all the other functions ;) Let's keep PRs focused on one thing and get them merged quickly. Then we can iterate if needed.

Reply:
Right. Just out of curiosity, does everything deprecated go away with a major release? Is it a formal part of the release process?